### ISSUE TITLE
WASAPI: allow bigger buffer size

### ISSUE THREAD

**Issue by SDLBugzilla** [libsdl-org/SDL#3498]
# This bug report was migrated from our old Bugzilla tracker.

**Reported in version:** HG 2.1
**Reported for operating system, platform:** Windows (All), All

# Comments on the original bug report:

On 2019-12-31 17:05:33 +0000, Vitaly Novichkov wrote:

> Hello!
> I need to stream a sound generated by the cycle-accurate chip emulator. It works fine on Linux and macOS and works with DirectSound and WinMM on Windows. However, on WASAPI it makes a very choppy result (chip emulator can't race with the shorter latency, it needs a bit bigger latency (at least 512 bytes)). WASAPI enforces to use a shorter latency than needed (approximately ~135).
> 
> Is here is possible to make some sort of software middle buffer to deal with heavy audio sources?
> 
> I don't know, should this patch help me or not https://bugzilla.libsdl.org/show_bug.cgi?id=4592, or it's totally unrelated thing.

On 2020-03-24 20:17:00 +0000, Ryan C. Gordon wrote:

> 
> We definitely _can_ buffer audio, and code exists to do this in SDL's audio loop, but I'm not sure if we force the sample size on WASAPI. I'll check.
> 
> --ryan.

On 2020-04-07 18:49:08 +0000, Ryan C. Gordon wrote:

> 
> Before we worry about buffering more, when you get a moment, can you see if the patch we just submitted from Bug # 4592 makes this any better for you?
> 
> --ryan.

**Comment by Wohlstand** [comment]
About this issue: once upon I found another thing at me: I had incorrectly bypassed the buffer size from my music libraries into the SDL Audio Stream, I had to pass bigger fragments than needed, which caused my library to load more to race with the time and get the requested buffer block being generated as fast as possible. I changed my code:
https://github.com/WohlSoft/SDL-Mixer-X/commit/b29ac96f5d8e6ff562fcb3fe31117a79be85c42f#diff-14cfc6ebeb28040589ab1331d8ea5f9803ee25eda01d96db169b8535ad043785
And the issue with a choppy playback has gone. However, debug builds still suck with a choppy playback.
